src/data_structures/suffix_array.rs

```diff
@@ -22,6 +22,10 @@ use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
 use data_structures::bwt::{DerefBWT, DerefLess, DerefOcc};
 
+// TODO: figure out how to make generic on `SuffixArray` type w/o `as` cast
```

Comment:
It's a little less ergonomic that what I originally intended (since functions using it have to be generic both on T and SA), but I think it'll do.